lib/computer_player.rb

```diff
@@ -0,0 +1,29 @@
+class ComputerPlayer
+  attr_reader :mark
+
+  def initialize(mark, board, display)
+    @board = board
+    @display = display
+    @mark = mark
+  end
+
+  def move
+    @display.make_move_message(@mark)
+    index = retrieve_computer_index
+    computer_is_thinking
+    @display.computer_move(index + 1)
```

Comment:
Why do you use index + 1? Does `retrieve_computer_index` return an index that isn't valid until you add 1?

Reply:
It is displaying to the user what position the computer chose. Im not sure if it is necessary, but it was part of the UI.